.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+### AUTO GENERATED. DO NOT MODIFY. ###
+# This file should be auto generated by the files in the src folder.
+# You can update it by running `yarn run circleci:update-config`.
+commands:
+    check_config:
+        description: This command builds the circle config from the files in src and validates that it is up-to-date and valid.
+        steps:
+            - run:
+                command: curl -fLSs https://raw.githubusercontent.com/CircleCI-Public/circleci-cli/master/install.sh | sudo bash
+                name: Install CircleCI CLI
+            - run:
+                command: yarn run circleci:update-config
+                name: Build circle config
+            - run:
+                command: |
+                    FILES_MODIFIED=""
+                    setcommit () {
+                      FILES_MODIFIED=$(git status -s | grep -i -E '.*circleci/config.yml')
+                    }
+                    setcommit || true
+                    if [ -z "$FILES_MODIFIED" ]
+                    then
+                      echo "The CircleCI config is up to date."
+                      exit 0;
+                    else
+                      echo "The CircleCI config is not up to date. You can update it by running `yarn run circleci:update-config`."
+                      exit 1;
+                    fi
+                name: CircleCI config up to date
```

Comment:
@klinzo Why is it needed, that the pipeline fails if dirty. Could also be ignored, because the pipeline builds an new config anyway. So why force to do it manually?

Reply:
The pipeline does not build a new config but just uses the config.yml. In order to avoid discrepancies between the actual config code in src and the stuff that is actually executed, this checks if both are the same.

Reply:
So what does the "Build circle config" step do? Can it be removed? Validation is done below anyway?